util/src/main/scala/scala/scalanative/io/VirtualDirectory.scala

```diff
@@ -98,9 +99,18 @@ object VirtualDirectory {
   private final class JarDirectory(path: Path)(implicit in: Scope)
       extends NioDirectory {
     private val fileSystem: FileSystem =
-      acquire(
-        FileSystems.newFileSystem(URI.create(s"jar:${path.toUri}"),
-                                  Map("create" -> "false").asJava))
+      acquire {
+        val uri = URI.create(s"jar:file:${path}")
+        Try(FileSystems.newFileSystem(uri, Map("create" -> "false").asJava)) match {
+          case Success(s) => s
+          case Failure(e) =>
+            e match {
+              case e: FileSystemAlreadyExistsException =>
+                FileSystems.getFileSystem(uri)
+              case _ => throw e
+            }
+        }
+      }
```

Comment:
`scala.uil.Try` might be an overkill here, normal `try catch` would do the job just as well.